docs/signature/README.md

```diff
@@ -0,0 +1,228 @@
+```
+
+An nv2 client would generate the following content to be signed:
+
+``` JSON
+{
+  "signed": {
+      "exp": 1626938793,
+      "nbf": 1595402793,
+      "iat": 1595402793,
+      "digest": "sha256:3351c53952446db17d21b86cfe5829ae70f823aff5d410fbf09dff820a39ab55",
+      "size": 528,
+      "references": [
+          "registry.example.com/hello-world:latest",
+          "registry.example.com/hello-world:v1.0"
+      ]
+  }
+```
+
+The signature of the above would be represented as:
+
+``` JSON
+{
+  "signature": {
+    "typ": "x509",
+    "sig": "uFKaCyQ4MtVHemfLVq5gYZyeiClS20tksXzP7hhpeqqjCNK9DiHnoDpkq91sutLqd1o6RCxpfFVuGXy20oqRu1/ZoXXAVC3y7lS6z/wqJ4VDBKSj/H6xyYn7pH3GE8GHR6kjFPqrGsl/OS4yYH2oNXEm9W8Pju2wC381+FCgf4LNf7k6u2Uf4Fb0/Fl40qzvr0m2Fv5pXtRY+wdJctqJb+t408VcXJkNj0U7xoOe0zUr3l1A6xLYqjd0ZY08JBQ8FQul0Vpxrmg0Xdtwd/wEolvia48lxD1x7yphW5bFvJOTd62rOJgd4uI7jYJF3ZLmwjY+geMk5e6Wkp5OyXGjXw==",
+    "alg": "RS256",
+    "x5c": [
+      "MIIDmzCCAoOgAwIBAgIUFSzsIT4/pKtGzywuZWWE7ydiLBIwDQYJKoZIhvcNAQELBQAwXTELMAkGA1UEBhMCQVUxEzARBgNVBAgMClNvbWUtU3RhdGUxITAfBgNVBAoMGEludGVybmV0IFdpZGdpdHMgUHR5IEx0ZDEWMBQGA1UEAwwNKi5leGFtcGxlLmNvbTAeFw0yMDA3MjIwMzA2MTBaFw0yMTA3MjIwMzA2MTBaMF0xCzAJBgNVBAYTAkFVMRMwEQYDVQQIDApTb21lLVN0YXRlMSEwHwYDVQQKDBhJbnRlcm5ldCBXaWRnaXRzIFB0eSBMdGQxFjAUBgNVBAMMDSouZXhhbXBsZS5jb20wggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEKAoIBAQDM0MNLy/f1SyRM0ZQu3AtJnCU3O5x8nnOeV1mySmZNr2SCqR8+jENAoKE5FrrSi2ffMnFPP/7DqGnbb9+b1nD9ucFNsI1iW7IrF/GlqOM7jJhUMNnOyatz8mddtQgXr3SZ9bigbc/lxuVGacvi64DewoWzMFr4ZMGq8ik7aDnHryUDwXJFE+KGNbsReO1ePqKmPiLvkLG4sBTqeTuCk+Grrr5t1COujwuFWfhMjmRfq34QGqUZ3SHJYXPzOAxgV3fCmBP9IgHuSv/b1udx5Htf1BV7WlARtXfE21xuA6FM1Gq0pANUhcRF39KJRu4/RBZBmAxg7ces8hrZWTQ4LTo/AgMBAAGjUzBRMB0GA1UdDgQWBBR2pI+c2dexlOZCXLy84Baqu8NR8DAfBgNVHSMEGDAWgBR2pI+c2dexlOZCXLy84Baqu8NR8DAPBgNVHRMBAf8EBTADAQH/MA0GCSqGSIb3DQEBCwUAA4IBAQCH2tChjmvs6/2acw+cJYWkEExdXMEyjdUvqEIcs7W7Ce32My7RcMtJxybtqjV+UVghEVUzq1pNf0Dt5FhFkC6BDHnHv2SIO9jq2TvfDUcJgMMgwSZdSaISmxk+iFD9Cll+RU8KgeoYSnwojOixTksyeBRi5rePdO5smz/n4Bd4ToluKaw42tdWhF4SMgx2Y1nlyHpFlkdUYtJ6D8rOvbVRGQaxo8Td3mWCWPMBYcGvjwO9ESCP1JAK+Z6WXD46JWilsIUd3Y+0NrfvOYKUdhLWuz9LrQ5060qi1pHfYBOTAbyXfnW97EB3TAuMtqBBe6h3VNw00c1p7qrilE1Of9uN"
+    ]
+  }
+}
+```
+
+### Signature Persisted within an OCI Artifact Enabled Registry
+
+Both values are persisted in a `signature.json` file. The file would be submitted to a registry as an Artifact with null layers.
+The `signature.json` would be persisted wthin the `manifest.config` object
+
+``` SHELL
+oras push \
+  registry.example.com/hello-world:v1.0 \
+  --manifest-config signature.json:application/vnd.cncf.notary.config.v2+json
+```
+
+Would push the following manifest:
+
+``` JSON
+{
+  "schemaVersion": 2,
+  "config": {
+    "mediaType": "application/vnd.cncf.notary.config.v2+json",
+    "size": 1906,
+    "digest": "sha256:c7848182f2c817415f0de63206f9e4220012cbb0bdb750c2ecf8020350239814"
+  },
+  "layers": []
+}
+```
+
+## *Signature* Property Descriptions
+
+- **`signed`** *object*
```

Comment:
I don't see a version number here that would prevent rollback attacks. `nbf` and `exp` would help mitigate this to some limited extent, but it wouldn't prevent them.

Reply:
Trishank, are you referring to a schema version? Or, a TUF version?

Reply:
A version number separate from all other version numbers, and which should be contained in any signed metadata file. If this version number is incremented every time there is a change in the metadata, then attackers cannot replay previous metadata files that should be obsolete, even if they haven't expired.

Reply:
ok, this is the update scenario we haven't yet scoped. We'll address this in phase2.